netbox/data_source_netbox_prefix.go

```diff
@@ -31,7 +39,26 @@ func dataSourceNetboxPrefixRead(d *schema.ResourceData, m interface{}) error {
 	api := m.(*client.NetBoxAPI)
 
 	cidr := d.Get("cidr").(string)
+	description := d.Get("description").(string)
+
+	if cidr == "" && description == "" {
+		return errors.New("Either a prefix or a description should be given.")
```

Comment:
Well, ideal would be that both options are optional and if both are given, it searches for BOTH things (with AND), whereas if only one is given, it searches for that one. Ideally, this would also apply to any search fields that are introduced later.

This might not necessarily make sense for this specific resource, but, generally speaking, searching for an intersection of multiple attributes is nice.

I would implement flexible filters (unless there are specific problems with that approach that I am not aware of). If you do not want to do that, use ExactlyOneOf.